File: src/main/java/io/lsdconsulting/lsd/distributed/interceptor/captor/http/derive/HttpStatusDeriver.java
package io.lsdconsulting.lsd.distributed.interceptor.captor.http.derive;

import org.springframework.http.HttpStatus;

import java.util.Optional;

public class HttpStatusDeriver {
    public String derive(final int code) {
        return Optional
                .ofNullable(HttpStatus.resolve(code))
                .map(HttpStatus::toString)
                .orElse(String.format("<unresolved status:%s>", code));
    }
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/interceptor/captor/http/derive/HttpStatusDeriver.kt
package io.lsdconsulting.lsd.distributed.interceptor.captor.http.derive

import org.springframework.http.HttpStatus

class HttpStatusDeriver {
    fun derive(code: Int) = HttpStatus.resolve(code)?.toString() ?: String.format("<unresolved status:%s>", code)
}